### PR TITLE
WIP upgrade dependencies, remove gulp

### PR DESCRIPTION
**Pull request by oluoluoxenfree**
## Steps to test/reproduce
_Please explain how to best reproduce the issue and/or test the changes locally (including the pages/URLs/views/states to review)._

## Show me
_Provide screenshots/animated gifs/videos if necessary._

# REMEMBER: Attach this PR to the Trello card

### CODE REVIEW

File: .stylelintrc.yml
Comment (jgerigmeyer) [inline .stylelintrc.yml:45]: @mirisuzanne I disabled all the rules that were throwing errors. It'd probably be better for a Sass person to selectively disable the ones we don't like, and fix the rest of the errors.

File: package.json
Comment (jgerigmeyer) [inline package.json:38]: I'm not actually sure what watcher commands are useful in practice. Generally I prefer more control as a dev, setting my editor to lint on save, and running tests/sassdoc manually when desired. @mirisuzanne What do you think?

**Review by jgerigmeyer — APPROVED** [review]
@mirisuzanne @stacyk @dvdherron This is ready for your review. I can resolve merge conflicts once it's ready to merge.

File: package.json
Comment (mirisuzanne) [reply package.json:38]: That's how I like it too, though I'm not sure how to set my editor to lint with the project rules.

It was @stacyk that requested a watcher, and might have more thoughts on it.

Comment (stacyk) [reply package.json:38]: I'm surprised to hear that I am the only one (of three of us so far) who likes a watcher. I don't know what else the question is here? If the watcher is optional, then wouldn't it be nice to keep it (if it is working)? 

I'm running to lunch, but will pull this branch and test it when I return.

Comment (jgerigmeyer) [reply package.json:38]: @stacyk Yeah I'm fine having a watcher -- I just don't know what watched files or tasks you would find useful. Right now it's linting all Sass on any change to a Sass file (but not auto-fixing; just showing errors/warnings), running Sass tests on changes to test files (should that run on any Sass file change?), and generating Herman docs on any change to Sass source files.

**Comment by stacyk** [comment]
[I want Accoutrement on Sass Modules](https://trello.com/c/9tyINy2o/23-i-want-accoutrement-on-sass-modules)

File: sass/animate/_easing.scss
Comment (stacyk) [inline sass/animate/_easing.scss:68]: When the CSSers spoke this week, I THINK we leaned towards the parameters being separated by a line break so it is easier to scan/see which changes were made, etc. I used to be a "everything all on line line" kind of person but have since found line breaks helpful in certain situations. 

What do you think about this moving forward @dvdherron @mirisuzanne? Is there a preference that we decided upon?

If so, is there a setting that we need @oluoluoxenfree  @jgerigmeyer for that to take place?

File: test/scale/_test-data.scss
Comment (stacyk) [inline test/scale/_test-data.scss:41]: I know I said parameters on their own line in an above comment, but this feels a lot less readable when we force each thing on a line here as opposed to what we had before. I don't know if anyone would agree with that, but throwing it out there incase we want to adjust something here. Thoughts @dvdherron @mirisuzanne?

File: sass/animate/_easing.scss
Comment (jgerigmeyer) [reply sass/animate/_easing.scss:68]: @stacyk If we want to use Prettier to auto-format, I don't think there's an option for this. If we want to ditch Prettier and only use Stylelint, then we have more control but will need to do more of the formatting ourselves.

Comment (mirisuzanne) [reply sass/animate/_easing.scss:68]: I like having parameters on their own lines, with trailing commas -- but I'm not sure if there's a setting for it? Seems like there must be somewhere, since David's local linter was changing this the other way…

File: test/scale/_test-data.scss
Comment (mirisuzanne) [reply test/scale/_test-data.scss:41]: Yeah, I tends to agree with you when the map only has a single key/value -- but I like multiline when there's more than one. I'm not sure if a linter will let us be that specific? If I had to choose just one, I'd probably stick with the expanded version, and then ignore that rule for specific lines/files.

Comment (dvdherron) [reply test/scale/_test-data.scss:41]: @stacyk I agree that sass function parameters being on separate lines is generally more readable. If I had to choose, I would choose the stacked approach. 

I agree that the maps are a lot less readable in the way shown here ☝🏽 . If there's a rule to keep map keys/values on the same line that would be preferred.

File: sass/animate/_easing.scss
Comment (jgerigmeyer) [reply sass/animate/_easing.scss:68]: Yes, there are stylelint rules for it, but only if we remove Prettier. That's your call.

https://github.com/prettier/stylelint-prettier

File: package.json
Comment (stacyk) [reply package.json:38]: > running Sass tests on changes to test files (should that run on any Sass file change?)

@jgerigmeyer I guess while working on this project specifically I wanted tests to run upon save on an scss file that is used in any test file. I realize that is complicated and maybe not how it works. For instance, if I made a change to `sass/layout/query-utils` the tests in the text/layout/query-utils would also run so I can see if the change I made in the main file made any difference to a test's results. It is a small enough project to run all sass tests if it is too tricky to figure out which files are `@use`d in a test. Does that make sense?

Comment (jgerigmeyer) [reply package.json:38]: Sure, that makes sense. I'll set it to run all tests on any change to any Sass file.

Comment (dvdherron) [reply package.json:38]: > 

my two cents: I kept thinking of having a watcher while working on tests. Sort of how we do it for JS testing in Metecho. other than that, linting manually works fine for me.

**Review by jgerigmeyer — APPROVED** [review]

**Pull request merged**
